src/Features/Core/Portable/QualifyMemberAccess/AbstractQualifyMemberAccessDiagnosticAnalyzer.cs

```diff
@@ -121,19 +110,20 @@ private void AnalyzeOperation(OperationAnalysisContext context, IOperation opera
             var optionValue = optionSet.GetOption(applicableOption, context.Operation.Syntax.Language);
 
             var shouldOptionBePresent = optionValue.Value;
-            var isQualificationPresent = IsAlreadyQualifiedMemberAccess(simpleName);
-            if (shouldOptionBePresent && !isQualificationPresent)
+            var severity = optionValue.Notification.Severity;
+            if (!shouldOptionBePresent || severity == ReportDiagnostic.Suppress)
```

Comment:
ah. so it is not checking diagnostic's supression but option's supression to see whether analyzer is turned off? is this option work with ruleset as well? I think ruleset's diagnostic severity comes from compilation option.

Reply:
> is this option work with ruleset as well? I think ruleset's diagnostic severity comes from compilation option.

It is my understanding that the presence of a value in a rule set file will force the severity of diagnostics to be overridden, so whatever we do here would no longer have an impact. With that said, our IDE analyzers do not work particularly well with rule set files; they were implemented with .editorconfig as the recommended configuration mechanism.

> by the way, is this work with command line case? is this another IDE only thing?

None of these analyzers are currently supported on the command line.